File: gplx-be/src/main/java/com/demo/entities/Question.java
package com.demo.entities;

import jakarta.persistence.*;

import java.util.List;

@Entity
@Table(name = "question")
public class Question {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Integer id;

    @Column(columnDefinition = "text")
    private String content;

    @Column(name = "is_failed")
    private boolean isFailed;

    @Column(columnDefinition = "text")
    private String image;

    @Column(name = "explain")
    private String explain;

    private boolean status;

    @Column(name = "is_rankA")
    private boolean isRankA;

    @OneToMany(mappedBy = "question")
    private List<Answer> answers;

    @OneToMany(mappedBy = "question")
    private List<TestDetails> testDetails;

    public Integer getId() {
        return id;
    }

    public void setId(Integer id) {
        this.id = id;
    }

    public String getContent() {
        return content;
    }

    public void setContent(String content) {
        this.content = content;
    }

    public boolean getIsFailed() {
        return isFailed;
    }

    public void setIsFailed(boolean isFailed) {
        this.isFailed = isFailed;
    }

    public String getImage() {
        return image;
    }

    public void setImage(String image) {
        this.image = image;
    }

    public boolean isStatus() {
        return status;
    }

    public void setStatus(boolean status) {
        this.status = status;
    }

    public List<Answer> getAnswers() {
        return answers;
    }

    public void setAnswers(List<Answer> answers) {
        this.answers = answers;
    }

    public List<TestDetails> getTestDetails() {
        return testDetails;
    }

    public void setTestDetails(List<TestDetails> testDetails) {
        this.testDetails = testDetails;
    }

    public boolean getIsRankA() {
        return isRankA;
    }

    public void setRankA(boolean rankA) {
        isRankA = rankA;
    }

    public String getExplain() {
        return explain;
    }

    public void setExplain(String explain) {
        this.explain = explain;
    }
}

File: gplx-fe/lib/entities/Question.dart
import 'package:gplx/entities/Answer.dart';

class Question {
  int? id;
  String? content;
  String? image;
  bool? failed; // Thay failed bằng is_failed
  String? explain; // Thêm trường explain
  bool? status;
  bool? isRankA;  // Thêm trường is_rankA
  List<Answer>? answers;

  Question({
    this.id,
    this.content,
    this.image,
    this.failed,
    this.explain,
    this.status,
    this.isRankA,
    this.answers,
  });

  // Tạo từ Map (JSON)
  factory Question.fromMap(Map<String, dynamic> map) {
    return Question(
      id: map['id'] as int?,
      content: map['content'] as String?,
      image: map['image'] as String?,
      failed: map['is_failed'] == 1, // Chuyển tinyint(1) thành boolean (1 = true, 0 = false)
      explain: map['explain'] as String?, // Thêm ánh xạ cho explain
      status: map['status'] == 1, // Chuyển tinyint(1) thành boolean
      isRankA: map['is_rankA'] == 1, // Chuyển tinyint(1) thành boolean
      answers: (map['answers'] as List<dynamic>?)
          ?.map((e) => Answer.fromMap(e))
          .toList(),
    );
  }

  // Chuyển thành Map (JSON)
  Map<String, dynamic> toMap() {
    return {
      'id': id,
      'content': content,
      'image': image,
      'is_failed': failed == true ? 1 : 0, // Chuyển boolean thành 1 hoặc 0
      'explain': explain, // Thêm explain vào map
      'status': status == true ? 1 : 0, // Chuyển boolean thành 1 hoặc 0
      'is_rankA': isRankA == true ? 1 : 0, // Chuyển boolean thành 1 hoặc 0
      'answers': answers?.map((e) => e.toMap()).toList(),
    };
  }
}